sotodlib/io/imprinter.py

```diff
@@ -1099,6 +1140,28 @@ def get_g3tsmurf_obs_for_book(self, book):
         )
         return {o.obs_id: o for o in obs}
 
+    def all_bound_until(self):
+        """report a datetime object to indicate that all books are bound
+        by this datetime.
+        """
+        session = self.get_session()
+        # sort by start time and find the start time by which
+        # all books are bound
+        books = session.query(Books).order_by(Books.start).all()
+        for book in books:
+            if book.status < BOUND:
+                return book.start
+        return book.start  # last book
+
+    def all_bound_until_hk(self):
```

Comment:
is there anything with HK here that you should also be looking at for smurf and stray books?

Reply:
hmm that's right, there's nothing special about hk. It's the same function for smurf or stray. I can remove this redundant function.